### PR TITLE
When checking read/write access, if there is no access, surface the e…

### PR DESCRIPTION
**Pull request by natalieparellano**
…rror that occurred

This can be helpful in debugging failed builds where access is expected

### CODE REVIEW

**Review by jabrown85 — APPROVED** [review]
LGTM

File: platform/run_image.go [hunk 1]
```diff
@@ -78,31 +78,23 @@ func BestRunImageMirrorFor(targetRegistry string, runImageMD files.RunImageForEx
 
 	// Select the first run image we have access to
 	for _, image := range runImageMirrors {
-		ok, err := checkReadAccess(image, keychain)
-		if err != nil {
-			return "", err
-		}
-		if ok {
+		if ok, _ := checkReadAccess(image, keychain); ok {
```
Comment (natalieparellano) [inline platform/run_image.go:81]: Ideally we'd log the error at this point, but that requires a logger, which is a bigger change

File: cmd/lifecycle/main.go [hunk 2]
```diff
@@ -143,7 +143,9 @@ func verifyReadAccess(imageRef string, keychain authn.Keychain) error {
 		return nil
 	}
 	img, _ := remote.NewImage(imageRef, keychain)
-	if !img.CheckReadAccess() {
+	canRead, err := img.CheckReadAccess()
+	if !canRead {
+		cmd.DefaultLogger.Debugf("Error checking read access: %s", err)
```
Comment (cmoulliard) [inline cmd/lifecycle/main.go:148]: We should log that without using Debug as some users using Tekton or kpack could not be aware about how to enable log level !!

File: cmd/lifecycle/main.go [hunk 3]
```diff
@@ -154,7 +156,9 @@ func verifyReadWriteAccess(imageRef string, keychain authn.Keychain) error {
 		return nil
 	}
 	img, _ := remote.NewImage(imageRef, keychain)
-	if !img.CheckReadWriteAccess() {
+	canReadWrite, err := img.CheckReadWriteAccess()
+	if !canReadWrite {
+		cmd.DefaultLogger.Debugf("Error checking read/write access: %s", err)
```
Comment (cmoulliard) [inline cmd/lifecycle/main.go:161]: We should log that without using Debug as some users using Tekton or kpack could not be aware about how to enable log level !!

**Pull request merged**